web/modules/custom/portland_openid_connect/src/Util/PortlandOpenIdConnectUtil.php

```diff
@@ -37,11 +37,16 @@ public static function removeUserFromGroup($account, $group_id)
       }
       $group_content->group_roles->appendItem(['target_id' => $role->id()]);
     }
-    if ($has_employee_role) $group_content->save();
-
-    // $group = \Drupal\group\Entity\Group::load($group_id);
-    // $group->removeMember($account);
-    // $group->save();
+    // If the user has no role in the group, remove the user completely
```

Comment:
@kxwang, wouldn't this break the automatic management of a private group since we don't have an employee role set up there yet. Related, I think we should add that role and private group syncing based on AD group.

I'm also wondering if a user that is added to a team or private group by an admin and not given a role would get stripped by this process. 🤔

Reply:
@joshuami The AD sync is based on the taxonomy https://employees.portland.gov/admin/structure/taxonomy/manage/group_ad_name_list/overview. There is no private group in the list. The sync only works with Employee Groups for now. In general, the AD group is at bureau level which makes it unlikely that it'll become a private group.
The foreach loop above also only checks if the user has membership in Employee Groups. See the role's machine name "employee-member", "employee-employee". It won't remove any membership in private groups. You can test it by adding a user to a private group and then mark it contact only. It should not change private group membership at all.

Reply:
I'll create a follow up story for using the employee role and syncing on private groups.